modules/gapi/src/backends/cpu/gcpuvideo.cpp

```diff
@@ -107,13 +107,119 @@ GAPI_OCV_KERNEL_ST(GCPUBackgroundSubtractor,
+                      std::shared_ptr<cv::KalmanFilter> &state, const cv::GCompileArgs&)
+    {
+        state = std::make_shared<cv::KalmanFilter>(kfParams.transitionMatrix.rows, kfParams.measurementMatrix.rows,
+                                                   kfParams.controlMatrix.cols, kfParams.transitionMatrix.type());
+
+        // initial state
+        state->statePost = kfParams.state;
+        state->errorCovPost = kfParams.errorCov;
+
+        // dynamic system initialization
+        state->controlMatrix = kfParams.controlMatrix;
+        state->measurementMatrix = kfParams.measurementMatrix;
+        state->transitionMatrix = kfParams.transitionMatrix;
+        state->processNoiseCov = kfParams.processNoiseCov;
+        state->measurementNoiseCov = kfParams.measurementNoiseCov;
+    }
+
+    static void run(const cv::Mat& measurements, bool haveMeasurement,
+                    const cv::Mat& control, const cv::gapi::video::KalmanParams&,
+                    cv::Mat &out, cv::KalmanFilter& state)
+    {
+        cv::Mat pre = cv::norm(control, cv::NORM_INF) != 0 ? state.predict(control) : state.predict();
+        haveMeasurement ? state.correct(measurements).copyTo(out) : pre.copyTo(out);
+    }
+};
+
+GAPI_OCV_KERNEL_ST(GCPUKalmanFilterNoControl, cv::gapi::video::GKalmanFilterNoControl, cv::KalmanFilter)
+{
+    static void setup(const cv::GMatDesc&, const cv::GOpaqueDesc&,
+                      const cv::gapi::video::KalmanParams& kfParams,
+                      std::shared_ptr<cv::KalmanFilter> &state,
+                      const cv::GCompileArgs&)
+    {
+        state = std::make_shared<cv::KalmanFilter>(kfParams.transitionMatrix.rows, kfParams.measurementMatrix.rows,
+                                                   0, kfParams.transitionMatrix.type());
+        // initial state
+        state->statePost = kfParams.state;
+        state->errorCovPost = kfParams.errorCov;
+
+        // dynamic system initialization
+        state->measurementMatrix = kfParams.measurementMatrix;
+        state->transitionMatrix = kfParams.transitionMatrix;
+        state->processNoiseCov = kfParams.processNoiseCov;
+        state->measurementNoiseCov = kfParams.measurementNoiseCov;
+    }
+
+    static void run(const cv::Mat& measurements, bool haveMeasurement,
+                    const cv::gapi::video::KalmanParams&, cv::Mat &out,
+                    cv::KalmanFilter& state)
+    {
+        cv::Mat pre = state.predict();
+        haveMeasurement ? state.correct(measurements).copyTo(out) : pre.copyTo(out);
+    }
+};
+
+namespace cv { namespace gapi { namespace video {
+void checkParams(const cv::gapi::video::KalmanParams& kfParams,
+                        const cv::GMatDesc& measurement, const bool controlCapable, const int type)
+{
+    GAPI_Assert(type == CV_32FC1 || type == CV_64FC1);
+
+    int depth = (type == CV_32FC1) ? CV_32F : CV_64F;
```

Comment:
You can also do it via macro if you wish:
` int depth = CV_MAT_DEPTH(type);`

Reply:
Done.